pkg/sink/kafka/metrics_collector.go

```diff
@@ -71,9 +77,13 @@ func NewSaramaMetricsCollector(
 }
 
 func (m *saramaMetricsCollector) Run(ctx context.Context) {
-	ticker := time.NewTicker(RefreshMetricsInterval)
+	// Initialize brokers.
+	m.updateBrokers(ctx)
+
+	refreshMetricsTicker := time.NewTicker(RefreshMetricsInterval)
+	refreshClusterMetaTicker := time.NewTicker(refreshClusterMetaInterval)
 	defer func() {
-		ticker.Stop()
+		refreshMetricsTicker.Stop()
```

Comment:
shall we also stop the `refreshClusterMetaTicker`  ?

Reply:
oops! Good catch.